src/couch_server.erl

```diff
@@ -549,9 +554,13 @@ delete_db_file(RootDir, FullFilePath, Options) ->
     end.
 
 rename_on_delete(Original) ->
+    file:rename(Original, deleted_filename(Original)).
+
+deleted_filename(Original) ->
     {{Y,Mon,D}, {H,Min,S}} = calendar:universal_time(),
+    UUID = ?b2l(couch_uuids:random()),
```

Comment:
I get what you intended but this seems awkward. Could delete_file instead return `ok | {renamed, NewName} | {error, Reason}` instead? Then we know where to look in the tests.